cliquet/storage/postgresql/schema.sql

```diff
@@ -26,13 +30,53 @@ CREATE TABLE IF NOT EXISTS records (
 
     PRIMARY KEY (id, parent_id, collection_id)
 );
+--
+-- CREATE INDEX IF NOT EXISTS will be available in PostgreSQL 9.5
+-- http://www.postgresql.org/docs/9.5/static/sql-createindex.html
+DO $$
+BEGIN
 
-DROP INDEX IF EXISTS idx_records_parent_id_collection_id_last_modified;
-CREATE UNIQUE INDEX idx_records_parent_id_collection_id_last_modified
+  IF NOT EXISTS (
+    SELECT 1 FROM pg_indexes
+       WHERE indexname = 'idx_records_parent_id_collection_id_last_modified'
+       AND tablename = 'records'
+  ) THEN
+
+  CREATE UNIQUE INDEX idx_records_parent_id_collection_id_last_modified
     ON records(parent_id, collection_id, last_modified DESC);
```

Comment:
Well, what I mean is that there are two ways to write the same behaviour.
For example, in python:

```python
if key in mydict:
    print mydict[key]
```
is equivalent (and less performant btw) to: 
```python
try:
    print mydict[key]
except KeyError:
    pass
```

So for the index creation I guess it's the same.
Either : 
```sql
DO $$
BEGIN
    IF NOT EXISTS ( ... )
    THEN
         CREATE INDEX
    END IF;
END$$;
```

or 


```sql
DO $$
BEGIN
    BEGIN
         CREATE INDEX ...
    EXCEPTION WHEN <error> THEN
            RAISE NOTICE 'Index exists with this name';
    END;
END$$;
```

Indeed, it would have prevented the typo you add in the commit you mentionned :)